### DISCUSSION TITLE
Custom validation based on two input values

### DISCUSSION THREAD

**Discussion by emilpaw** [react-hook-form/react-hook-form#7252] [Q&A]
Hello,

I have a custom component that sets to fields by using setValue(). Now I want to add a validation that works on both of the fields at once. To be specific, my two fields are both arrays and I want to show an error if there is no entry added to either one of the arrays.

To help make my use case clear I created a CSB example: https://codesandbox.io/s/react-hook-form-validation-for-two-fields-qxvfx?file=/src/App.tsx

So basically I want to set the form to be invalid if (data1.length + data2.length < 1).

I know there is a setError function I could maybe use, but I have two give it an input name which means I need two either give it "data1" or "data2". That's ugly I think. Is there a way I could create a FieldError without having an actual input field backing it?

Thanks in advance.

**Comment by bluebill1049** [comment]
`validate` with `getValues`?

**Reply by emilpaw** [reply]
As seen in the example I don't use `register`, can I still somehow use `validate`?

**Reply by bluebill1049** [reply]
you need to register, if you don't have input, you can still `register` at useEffect.

**Reply by emilpaw** [reply]
Thank you very much. I am much closer to solving my issue now.

Didn't think about using `register` without an input. I implemented your suggestion in my example: https://codesandbox.io/s/react-hook-form-validation-for-two-fields-forked-5rs65?file=/src/App.tsx

Two problems still remain:
1. I have registered both data1 and data2. Is there a way to combine these two field validations into one?
2. The field is not revalidated on input change and only on pressing submit again. Do I need to trigger validation manually?
